external/anomaly/ote_anomalib/callbacks/inference.py

```diff
@@ -63,17 +65,36 @@ def on_predict_epoch_end(self, _trainer: pl.Trainer, pl_module: AnomalyModule, o
             dataset_item.append_labels([ScoredLabel(label=label, probability=float(probability))])
             if self.task_type == TaskType.ANOMALY_DETECTION:
                 dataset_item.append_annotations(
-                    annotations=create_detection_annotation_from_anomaly_heatmap(
+                    create_detection_annotation_from_anomaly_heatmap(
                         hard_prediction=pred_mask,
                         soft_prediction=anomaly_map,
                         label_map=self.label_map,
                     )
                 )
+                # TODO: only add full normal label if dataset purpose is inference
+                if len(dataset_item.get_annotations()) == 1:
```

Comment:
What does that condition mean? Does it assume that annotation is empty before inference? Why?